include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -410,6 +409,17 @@ def ControlMergeOp : Handshake_Op<"control_merge", [
 
   let arguments = (ins Variadic<AnyType> : $dataOperands);
   let results = (outs AnyType : $result, AnyType : $index);
+  
+  let builders = [OpBuilder<
+    (ins "ValueRange":$operands), [{
+      assert(!operands.empty() && "cmerge needs at least one operand");
+      $_state.addOperands(operands);
+      // By default, the index result has an Index type
+      SmallVector<Type> resultTypes { operands[0].getType(), 
+                                      $_builder.getIndexType()};
+      $_state.addTypes(resultTypes);
```

Comment:
NIT: Is itppssible to build an ArrayRef here, instead of building a SmallVector? Alternatively, you could also use two calls to `addType`